docs/examples/201/vm-new-or-existing-conditions/main.bicep

```diff
@@ -0,0 +1,272 @@
+  name: publicIpName
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+    dnsSettings: {
+      domainNameLabel: publicIpDns
+    }
+  }
+}
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2019-08-01' = if (virtualNetworkNewOrExisting == 'new') {
+  name: networkSecurityGroupName
+  location: location
+  properties: {
+    securityRules: [
+      {
+        name: 'default-allow-22'
+        properties: {
+          priority: 1000
+          access: 'Allow'
+          direction: 'Inbound'
+          destinationPortRange: '22'
+          protocol: 'Tcp'
+          sourceAddressPrefix: '*'
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2017-09-01' = if (virtualNetworkNewOrExisting == 'new') {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: addressPrefixes
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+          networkSecurityGroup: {
+            id: nsg.id
+          }
+        }
+      }
+    ]
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2017-09-01' = {
+  name: nicName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: resourceId(virtualNetworkResourceGroupName, 'Microsoft.Network/virtualNetworks/subnets/', virtualNetworkName, subnetName)
+          }
+          publicIPAddress: any(!(publicIpNewOrExisting == 'none') ? publicIpAddressId : null)
+        }
+      }
+    ]
+  }
+  dependsOn: [
+    publicIp
+    vnet
+  ]
+}
+
+resource vm 'Microsoft.Compute/virtualMachines@2017-03-30' = {
+  name: vmName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    osProfile: {
+      computerName: vmName
+      adminUsername: adminUsername
+      adminPassword: adminPasswordOrKey
+      linuxConfiguration: any((authenticationType == 'password') ? null : linuxConfiguration)
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'Canonical'
+        offer: 'UbuntuServer'
+        sku: '16.04-LTS'
+        version: 'latest'
+      }
+      osDisk: {
+        caching: 'ReadWrite'
+        createOption: 'FromImage'
+      }
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic.id
+        }
+      ]
+    }
+    diagnosticsProfile: {
+      bootDiagnostics: {
+        enabled: true
+        storageUri: reference(resourceId(storageAccountResourceGroupName, 'Microsoft.Storage/storageAccounts/', storageAccountName), '2017-06-01').primaryEndpoints.blob
+      }
+    }
+  }
+  dependsOn: [
+    storageAccount
+  ]
```

Comment:
Is this actually going to work in the scenario that the storageAccount isn't deployed by this template?

Reply:
It should work according to https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/conditional-resource-deployment#runtime-functions.